_published/0012-authentication.md

```diff
@@ -0,0 +1,167 @@
+   avoid having to fill the development environment with --force-unsecure
+   bindings we should consider implementing certificate generation and importing
+   for dev environments.
+
+Implementations stages
+
+1. Docker compose setup for Hydra Proxy setup
+   - This involves integration with the currently existing DB.
+2. Hydra (Authorisation server) - Connect to IDP (Config for Azure AD iver.se)
+   - Config instructions documented [here](https://www.ory.sh/hydra/docs/reference/configuration)
+   - [OpenID-Configuration](https://login.microsoftonline.com/common/.well-known/openid-configuration)
+   -
+3. Wharf-web - configure and setup [angular-auth-oicd-client](https://github.com/damienbod/angular-auth-oidc-client)
+   - Implement login endpoint
+   - Implement consent endpoint
+   - Implement logout endpoint
+
+4. Wharf-api - Implement and test
+   - [Go SDK for ory Hydra](https://www.ory.sh/hydra/docs/sdk/go)
+   - Inspiration for [gin middleware here](https://github.com/janekolszak/gin-hydra/blob/master/ginhydra.go)
+     note this is outdated.
+
+7. Edit wharf-web services to use bearer tokens through angular middleware
+   configured through `agular-auth-oicd-client`
+
+8. Wharf-provider-* - Implement and test
+   - Likely identical to procedure for wharf-api.
+
+### Flows
+
+OpenID Connect - Flows
+A good guide on flows is available [here on auth0](https://auth0.com/docs/flows).
+Or for a more fun version checkout [this game by auth0](https://iverab-my.sharepoint.com/:i:/g/personal/fredrik_dyrvold_iver_se/Ef41bR71qeNOk-QWhCar0V4BZN8WqvM7iEf5g7ml9_s7Ew?e=ZoZRN6)
+
+### Identity provider
+
+We should connect to azure ad on our local deployments.
+
+We should not hav our own identity provider setup with wharf for now.
+
+We should also be able to publish code in such a way that the IDP is
+interchangeable. If it is used by the public they can set up their own IDP
+and connect to that.
+
+### Authorisation server
+
+There are many [alternatives](https://openid.net/developers/certified/) on
+alternative authorisation servers.
+
+We had a quick info share from spark the spark project. They are using open
+identity server 3 or 4. This however appears to be relatively tightly coupled
+with the .NET framework.
+
+One that seems to align well with this
+project is `Ory - Hydra` [github](https://github.com/ory/hydra)
+and [docs](https://www.ory.sh/hydra/docs/). The descriptions of them from their
+github at the present time is:
+
+```
+ORY Hydra is a hardened, OpenID Certified OAuth 2.0 Server and OpenID Connect
+Provider optimized for low-latency, high throughput, and low resource
+consumption. ORY Hydra is not an identity provider (user sign up, user
+login, password reset flow), but connects to your existing identity provider
+through a login and consent app. Implementing the login and consent app in a
+different language is easy, and exemplary consent apps (Node) and SDKs for all
+common languages are provided.
+```
```

Comment:
nit: Suggest using blockquote instead of codeblock.

I personally like to add the source link at the end of the citation instead of before it, as I find it easier to see where it's taken from.

```suggestion
project is "Ory - Hydra" ([docs](https://www.ory.sh/hydra/docs/)).
The descriptions of them from their GitHub at the present time is:

> ORY Hydra is a hardened, OpenID Certified OAuth 2.0 Server and OpenID Connect
> Provider optimized for low-latency, high throughput, and low resource
> consumption. ORY Hydra is not an identity provider (user sign up, user
> login, password reset flow), but connects to your existing identity provider
> through a login and consent app. Implementing the login and consent app in a
> different language is easy, and exemplary consent apps (Node) and SDKs for
> all common languages are provided.
>
> <https://github.com/ory/hydra>
```